docs/transformer.md

```diff
@@ -288,63 +288,79 @@ To generate a tutorial:
 
 1. A Killercoda maintainer reviews the PR to ensure that the generate tutorial works as expected.
 
-## Scripts and extra course files
-
-If your tutorial requires scripts or extra files, make sure to manually add them to the tutorial directory in the Killercoda repository. For example, if your tutorial requires a script to run:
-
-1. Add the bash script to the tutorial directory in the Killercoda repository. Refer to the [what-is-loki](../loki/what-is-loki/) tutorial for an example.
-2. Add the script to the desired step within the `index.json` file. Note that `foreground` scripts run in the foreground (seen in terminal), and `background` scripts run in the background (run in background thread). For example:
-   ```json
-   {
-      "title": "What is Loki?",
-      "description": "A sandbox enviroment to introduce Loki to new users.",
-      "details": {
-         "intro": {
-         "text": "intro.md",
-         "foreground": "script1.sh"
-         },
-         "steps": [
-         {
-            "text": "step1.md",
-            "foreground": "script2.sh"
-         }
-         ],
-         "finish": {
-         "text": "finished.md"
-         }
-      },
-      "backend": {
-         "imageid": "ubuntu"
-      }
-   }
-
-   ```
-
-For extra assets, such as images or configuration files:
-1. Create a directory called `assets` in the tutorial directory.
-2. Add the assets to the `assets` directory.
-3. Add the mount path to the `index.json` file:
-   ```json
-   {
-      "title": "Grafana Basics",
-      "description": "In this demo learn how to install and configure Grafana",
-      "details": {
-         "intro": {
-         "text": "intro.md"
-         },
-         "steps": [],
-         "finish": {
-         "text": "finished.md"
-         },
-         "assets": {
-         "host01": [
-            {"file": "*", "target": "/education"}
-         ]
-         }
-      },
-      "backend": {
-         "imageid": "ubuntu"
-      }
-   }
-   ```
-   Refer to the [grafana-basics](../grafana/grafana-basics/index.json) tutorial for an example.
\ No newline at end of file
+## Adding foreground and background scripts (experimental)
+
+> Note: It is recommended to use `foreground` scripts rather than `background` scripts. This allows the user to see the output of the script. 
+
+Foreground and background scripts are shell scripts that run during the introduction and finish pages of a tutorial. The scripts are useful for setting up the environment for the tutorial and cleaning up after the tutorial. Scripts are another asset that needs to be maintained, so use them sparingly. A good example of using a script is to update the `docker compose` package before running the tutorial.
```

Comment:
I think documenting the difference after introducing them is 
more valuable than a note about the difference before a reader knows what they are

```suggestion
Foreground and background scripts are shell scripts that run during the introduction and finish pages of a tutorial.
The scripts are useful for setting up the environment for the tutorial and cleaning up after the tutorial.

Scripts are another asset that needs to be maintained, so use them sparingly.
A good example of using a script is to update the Docker Compose package before running the tutorial.

Use foreground scripts when you want to the user to see the script output.
Use background scripts when you want to hide the output of the script.
```